"""fixed Quote.asset_id foreign key glitch

Revision ID: b628540fb548
Revises: 6982b700c427
Create Date: 2020-03-25 11:23:53.856219

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = 'b628540fb548'
down_revision = '6982b700c427'
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.drop_index('ix_quote_id_cmc', table_name='quote')
    op.drop_constraint('quote_asset_id_fkey', 'quote', type_='foreignkey')
    op.create_foreign_key(None, 'quote', 'asset', ['id_cmc'], ['id_cmc'])
    op.drop_column('quote', 'asset_id')
    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    op.add_column('quote', sa.Column('asset_id', sa.INTEGER(), autoincrement=False, nullable=True))
    op.drop_constraint(None, 'quote', type_='foreignkey')
    op.create_foreign_key('quote_asset_id_fkey', 'quote', 'asset', ['asset_id'], ['id'])
    op.create_index('ix_quote_id_cmc', 'quote', ['id_cmc'], unique=False)
    # ### end Alembic commands ###
